op-e2e/actions/upgrades/holocene_fork_test.go

```diff
@@ -127,3 +132,87 @@ func TestHoloceneLateActivationAndReset(gt *testing.T) {
 	requirePreHoloceneActivationLogs(e2esys.RoleSeq, 4)
 	requirePreHoloceneActivationLogs(e2esys.RoleVerif, 4)
 }
+
+func TestHoloceneInvalidPayload(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+	env := helpers.SetupEnv(t, helpers.WithActiveGenesisFork(rollup.Holocene))
+	ctx := context.Background()
+
+	requireDepositOnlyLogs := func(role string, expNumLogs int) {
+		t.Helper()
+		recs := env.Logs.FindLogs(testlog.NewMessageContainsFilter("deposits-only attributes"), testlog.NewAttributesFilter("role", role))
+		require.Len(t, recs, expNumLogs)
+	}
+
+	// Start op-nodes
+	env.Seq.ActL2PipelineFull(t)
+
+	// generate and batch buffer two empty blocks
+	env.Seq.ActL2EmptyBlock(t) // 1 - genesis is 0
+	env.Batcher.ActL2BatchBuffer(t)
+	env.Seq.ActL2EmptyBlock(t) // 2
+	env.Batcher.ActL2BatchBuffer(t)
+
+	// send and include a single transaction
+	env.Alice.L2.ActResetTxOpts(t)
+	env.Alice.L2.ActSetTxToAddr(&env.DeployParams.Addresses.Bob)
+	env.Alice.L2.ActMakeTx(t)
+
+	env.Seq.ActL2StartBlock(t)
+	env.SeqEngine.ActL2IncludeTx(env.Alice.Address())(t)
+	env.Seq.ActL2EndBlock(t) // 3
+	env.Alice.L2.ActCheckReceiptStatusOfLastTx(true)(t)
+	l2Unsafe := env.Seq.L2Unsafe()
+	const invalidNum = 3
+	require.EqualValues(t, invalidNum, l2Unsafe.Number)
+	b, err := env.SeqEngine.EthClient().BlockByNumber(ctx, big.NewInt(invalidNum))
+	require.NoError(t, err)
+	require.Len(t, b.Transactions(), 2)
+
+	// buffer into the batcher, invalidating the tx via signature zeroing
+	env.Batcher.ActL2BatchBuffer(t, func(block *types.Block) {
+		// Replace the tx with one that has a bad signature.
+		txs := block.Transactions()
+		newTx, err := txs[1].WithSignature(env.Alice.L2.Signer(), make([]byte, 65))
+		require.NoError(t, err)
+		txs[1] = newTx
+	})
+
+	// generate two more empty blocks
+	env.Seq.ActL2EmptyBlock(t) // 4
+	env.Seq.ActL2EmptyBlock(t) // 5
+	require.EqualValues(t, 5, env.Seq.L2Unsafe().Number)
+
+	// submit it all
+	env.ActBatchSubmitAllAndMine(t)
+	env.Miner.ActL1SafeNext(t) // needed?
```

Comment:
```suggestion
```

Not needed. Unless you mean to also finalize the L1 later. The "safe" label on L1 is the beaconchain justified level, a step towards finality. Not used by the op-stack, as it doesn't have any helpful guarantees.